homeassistant/components/google_generative_ai_conversation/entity.py

```diff
@@ -480,3 +489,64 @@ def create_generate_content_config(self) -> GenerateContentConfig:
                 ),
             ],
         )
+
+
+async def async_prepare_files_for_prompt(
+    hass: HomeAssistant, client: Client, files: list[Path]
+) -> list[File]:
+    """Append files to a prompt.
+
+    Caller needs to ensure that the files are allowed.
+    """
+    prompt_parts: list[File] = []
+
+    def append_files_to_prompt():
+        for filename in files:
+            if not Path(filename).exists():
+                raise HomeAssistantError(f"`{filename}` does not exist")
+            mimetype = mimetypes.guess_type(filename)[0]
+            with open(filename, "rb") as to_upload_file:
```

Comment:
I believe something like this also, since `filename` is a `Path`

```suggestion
            with filename.open("rb") as to_upload_file:
```